### ISSUE TITLE
{{ baseUrl }} not working with <img> tag when included via div id

### ISSUE THREAD

**Issue by damithc** [MarkBind/markbind#803]
v2.1.0

I have the following code in a subsite `<img src="{{baseUrl}}/about/images/chromeSaveAsPdf.png" />`

This used to work before, but now the link is broken.

Output https://nus-te3201.github.io/2019/admin/te3201-admin.html

![image](https://user-images.githubusercontent.com/1673303/55471118-a0106900-563b-11e9-9c5f-3f87f95b0bd0.png)

Upon further investigation, it seems the img URL is not working if it is in a file that is included using a div id e.g., 
`<include src="../book/about/usage.md#saving" />`:  Doesn't work
`<include src="../book/about/usage.md" />`:  Works

**Comment by damithc** [comment]
@sijie123 can look into this?

**Comment by sijie123** [comment]
Hi Prof,
Bug confirmed - thanks for reporting the bug. I have independently checked out the TE3201 module page and confirm that the URL rewrite had gone wrong.

I investigated into the issue, but oddly, I cannot replicate this bug.
My test setup involves a very similar procedure:

index.md:
```html
<include src="folder/sub.md#test" />
```
sub.md:
```html
<div id="test">
  <img src="{{baseUrl}}/folder/img/picture.png" />
</div>
```
MarkBind builds correctly and the image is displayed correctly.

Confused, I continued printing the image URL before it was processed. In all the testing I've done so far, the URL is usually untouched . It's usually `{{baseUrl}}/images/...`. 
In this case, something caused the URL to become `{{hostBaseUrl}}/images/...`.

The `{{hostBaseUrl}}` was something I didn't expect. Right now, the assumption is that URLs starting with "{{baseUrl}}" or "/" are treated as absolute references. Hence `{{hostBaseUrl}}` was treated as a relative reference.

I'm not too sure why the URL became `{{hostBaseUrl}}`. This behaviour was present before the relative src commit was introduced. (Maybe @Xenonym / @yamgent can advise why this happens?)
It's odd to me that this only happens when a file is included with a div tag. In fact, my short test above couldn't replicate the behaviour that Prof encountered. Maybe it might also have something to do with the "../".

To fix the issue immediately, I can create a patch to fix all behaviour involving `{{hostBaseUrl}}`, but we still need to figure out why only certain files were converted to this `{{hostBaseUrl}}`, and if any other such references exist.
